website/docs/r/cloudfront_distribution.html.markdown

```diff
@@ -142,6 +142,100 @@ resource "aws_cloudfront_distribution" "s3_distribution" {
 }
 ```
 
+### CloudFront distribution with custom origin
+
+The example below uses an S3 static web hosting origin for use-cases beyond asset downloads. AWS doesn't allow using the S3 website endpoint from CloudFront. This example works using a custom authentication and a custom origin to further provide caching and HTTPS.
+
+```hcl
+locals {
+  origin_id = "S3-Website-${aws_s3_bucket.website-bucket.bucket}"
+  cloudfront-authentication-user-agent = "somethingRandomEen8Ohta"
+}
+
+resource "aws_s3_bucket" "website-bucket" {
+  bucket = "webhosting-bucket-via-cloudfront"
+
+  website {
+    index_document = "index.html"
+  }
+}
+
+resource "aws_s3_bucket_policy" "bucket-policy" {
+  bucket = "${aws_s3_bucket.website-bucket.id}"
+
+  policy = <<POLICY
```

Comment:
Related to above suggestion, use the generated JSON from the `aws_iam_policy_document` rather than inline JSON.

```suggestion
  policy = "${data.aws_iam_policy_document.bucket-policy.json}"
```